### PR TITLE
Refactoring api into model adaptor and storage adaptor classes

### PR DESCRIPTION
**Pull request by zmhassan**
## Description

Refactoring code base. Cleaner api's and as the project is maturing it was agreed that we should break up monolithic anomaly_detector.py into clean api's and pluggable code that anyone could integrate models in academia with LAD.

### CODE REVIEW

**Comment by zmhassan** [comment]
@durandom I know this has some coala errors and I'll fix that. In the mean time, please review the code. @MichaelClifford We had a design discussion this morning and was great to get your feedback. Would like get your review. I'm going to add the preprocessing customization in a next PR. A wise engineer told me to keep my PR's small so we can get them reviewed faster/merged faster @durandom 👀. Let me know!

File: anomaly_detector/adaptors/base_model_adaptor.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"""Base Model Adaptor Class - As we try different models and developers can contribute their custom code."""
+
+from abc import ABCMeta, abstractmethod
```
Comment (durandom) [inline anomaly_detector/adaptors/base_model_adaptor.py]: is abc a python thing?

Comment (zmhassan) [reply anomaly_detector/adaptors/base_model_adaptor.py]: yea it is. its part of python-3  https://www.python.org/dev/peps/pep-3119/

**Comment by zmhassan** [comment]
> can you change the test in this PR too, so that it tests the new AnomalyDetectorFacade?

Yes. It will be easy to swap the old with the new.

**Comment by zmhassan** [comment]
@durandom It's all fixed. I will rebase and if this PR is good please merge.

File: anomaly_detector/anomaly_detector_facade.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+class AnomalyDetectorFacade:
+    """For external interface for integration different adapters for custom models and training logic."""
+
+    def __init__(self, config):
+        """Abstraction around model adapter run method."""
+        storage_adapter = SomStorageAdapter(config)
+        model_adapter = SomModelAdapter(storage_adapter)
+        self.__model_adapter = model_adapter
```
Comment (zmhassan) [inline anomaly_detector/anomaly_detector_facade.py]: I could reduce this two lines 13,14 into 1 line: 
```        self.__model_adapter =SomModelAdapter(storage_adapter)```

**Comment by zmhassan** [comment]
@durandom This PR is ready for review. If this is good please merge.

**Pull request merged**